### PR TITLE
houdini: 17.0.352 -> 17.5.327

### PR DESCRIPTION
**Pull request by emptyflask**
###### Motivation for this change
Houdini 17.5 was released in March 2019, just bringing it up to date.

###### Things done
Houdini still depends on this [fix](https://github.com/NixOS/nixpkgs/issues/66544) by @ambrop72, so it's important that gets merged as well.

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/applications/misc/houdini/runtime.nix [hunk 1]
```diff
@@ -29,11 +29,11 @@ let
   license_dir = "~/.config/houdini";
 in
 stdenv.mkDerivation rec {
-  version = "17.0.352";
+  version = "17.5.327";
   pname = "houdini-runtime";
   src = requireFile rec {
     name = "houdini-${version}-linux_x86_64_gcc6.3.tar.gz";
```
Comment (c0bw3b) [inline pkgs/applications/misc/houdini/runtime.nix]: Still called `...._gcc6.3.tar.gz` with the bump to 17.5 ?

**Review by c0bw3b — CHANGES_REQUESTED** [review]
Could you also change https://sidefx.com to https://www.sidefx.com in message and meta.homepage please?
The former leads to a TLS certificate error.

File: pkgs/applications/misc/houdini/runtime.nix (hunk 1 above)
Comment (emptyflask) [reply pkgs/applications/misc/houdini/runtime.nix]: Yes, still gcc6.3

**Review by c0bw3b — APPROVED** [review]

**Comment by c0bw3b** [comment]
Thanks for the contribution to nixpkgs @emptyflask :)

**Comment by PlumpMath** [comment]
lib.makeLibraryPath [
                          nss nspr expat libxcb libXScrnSaver pciutils libpciaccess #xorg
                          libglvnd
                          cairo libdrm wayland  wayland-protocols  openssl libffi
                          libomxil-bellagio libva-minimal libva flex libelf libvdpau
                          mesa mesa_glu
                          vaapiIntel libvdpau-va-gl vaapiVdpau intel-ocl 
                          # add houdini-18

...
**I'd like to houdini-18.**

--------------------------
ERROR:
Crash report; Unknown App Version 18.0.287 [linux-x86_64-gcc6.3]
Uptime 0 seconds
Sun Dec  1 22:37:28 2019
Caught signal 6

Traceback from 5757 ThreadId=0x74ee21626b40
AP_Interface::coreDumpChaser(UTsignalHandlerArg) <libhoudiniui.so>
AP_Interface::si_CrashHandler::chaser(UTsignalHandlerArg) <libhoudiniui.so>
signalCallback(UTsignalHandlerArg) <libhoudiniut.so>
UT_Signal::UT_ComboSignalHandler::operator()(int, siginfo_t*, void*) const <libhoudiniut.so>
UT_Signal::processSignal(int, siginfo_t*, void*) <libhoudiniut.so>
__funlockfile <libpthread.so.0>
__GI_raise <libc.so.6>
__GI_abort <libc.so.6>
QMessageLogger::fatal(char const*, ...) const <libqt5core.so.5>
QGuiApplicationPrivate::createPlatformIntegration() <libqt5gui.so.5>
QGuiApplicationPrivate::createEventDispatcher() <libqt5gui.so.5>
QCoreApplicationPrivate::init() <libqt5core.so.5>
QGuiApplicationPrivate::init() <libqt5gui.so.5>
QApplicationPrivate::init() <libqt5widgets.so.5>
QDialog::sizeHint() const <hkey-bin>
QDialog::sizeHint() const <hkey-bin>
__libc_start_main <libc.so.6>
QDialog::sizeHint() const

**Comment by emptyflask** [comment]
@PlumpMath Looks like you might have gotten further with v18.0 than me, this is what I ran into. I also sent an email to SideFX to see if they had any insight.

After adding some dependencies that the binary appeared to require (nss, nspr, expat, xorg.libXScrnSaver, libxcb, pciutils), I ended up getting this error message:

```
    Qt Error: This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

    Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, webgl, xcb.

    Qt Error: This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

    Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, webgl, xcb.

    18323:  (sent by pid 18323)
    Crash log saved to /tmp/houdini_temp/crash.jon_18323_log.txt
```
```
    Crash report from jon; Unknown App Version 18.0.287 [linux-x86_64-gcc6.3]
    Uptime 0 seconds
    Sun Dec  1 21:09:24 2019
    Caught signal 6
    Traceback from 18323 ThreadId=0x7fd666adfb40
    AP_Interface::coreDumpChaser(UTsignalHandlerArg)
    AP_Interface::si_CrashHandler::chaser(UTsignalHandlerArg)
    signalCallback(UTsignalHandlerArg)
    UT_Signal::UT_ComboSignalHandler::operator()(int, siginfo_t*, void*) const
    UT_Signal::processSignal(int, siginfo_t*, void*)
    __funlockfile
    __GI_raise
    __GI_abort
    QMessageLogger::fatal(char const*, ...) const
    QGuiApplicationPrivate::createPlatformIntegration()
    QGuiApplicationPrivate::createEventDispatcher()
    QCoreApplicationPrivate::init()
    QGuiApplicationPrivate::init()
    QApplicationPrivate::init()
    QDialog::sizeHint() const
    QDialog::sizeHint() const
    __libc_start_main
    QDialog::sizeHint() const  
```

**Comment by emptyflask** [comment]
SideFX responded with:

> Houdini 18 bumped up its Qt version to 5.12 which has a new package dependency on Linux.
> 
> If you install the "libxkbcommon-x11" package (might be named slightly differently on your OS), then I believe that would fix the startup problems.
> 
> The new package dependency is subtly mentioned on our System Requirements page:
> https://www.sidefx.com/Support/system-requirements/

I haven't had a chance to look into it any further yet...

**Comment by PlumpMath** [comment]
What should I do? Other operating system tutorials are good, so some more
pleasee... is there a good workaround reference & tip?

-,.- py3 coming~*

2019년 12월 3일 (화) 오전 3:40, Jon Roberts <notifications@github.com>님이 작성:

**Pull request merged**